rust/frontend/tvm-sys/build.rs

```diff
@@ -0,0 +1,45 @@
+extern crate bindgen;
+
+use std::env;
+use std::error::Error;
+use std::path::PathBuf;
+use std::process;
+use std::result::Result;
+
+const TVM_RUNTIME: &'static str = "tvm_runtime";
+
+fn main() {
+    match run() {
+        Ok(_) => (),
+        Err(err) => {
+            eprintln!("error occured during the build: {:?}", err);
+            process::exit(1)
+        }
+    }
+}
+
+fn run() -> Result<(), Box<dyn Error>> {
+    println!("cargo:rustc-link-lib=dylib={}", TVM_RUNTIME);
+    let lib = format!("lib{}", TVM_RUNTIME);
+    println!("cargo:rustc-link-search=native={}", lib);
+    let tvm_home = env::var("TVM_HOME").expect("TVM_HOME not found!");
+    let bindings = bindgen::Builder::default()
+        .header(format!("{}/include/tvm/runtime/c_runtime_api.h", tvm_home))
+        .clang_arg(format!("-I{}/3rdparty/dlpack/include/", tvm_home))
+        .rustified_enum("DLDataTypeCode")
```

Comment:
the bindgen docs recommend against rustified enums https://docs.rs/bindgen/0.45.0/bindgen/struct.Builder.html#method.rustified_enum

Reply:
right! I agree but it seems the [blocker](https://github.com/rust-lang/rust/issues/36927) is related to non exhaustive which doesn't concern us here! Also since I'm wrapping some enums, I'll have to copy the members one of one and is just ugly.

Reply:
>  doesn't concern us here

sure it does. what happens if you try matching on the rustified enum, the runtime_api.h enum has a field added or rmoved, and suddenly the code doesn't compile.

Reply:
Regarding to `Use this with caution. You should not be using Rust enums unless you have complete control of the C/C++ code that you're binding to`
Don't we have control over the c api? I would agree if this is a binding for a third party lib only having the runtime share lib and the c api, but this is within TVM. Actually [tensorflow rust](https://github.com/tensorflow/rust/blob/master/tensorflow-sys/src/bindgen.rs#L186) uses the rustied enums a lot.

Reply:
Fair enough! it turns out that I don't need the `rustified_enum` as I've already rewritten them in types. It was just minor changes for `TVMDeviceType`.